app/services/news_service.py

```
@@ -35,4 +37,31 @@ async def fetch_news(
+                    article["id"] = str(uuid.uuid4())
+                    articles_to_cache[article["id"]] = article.copy()
+
+        # cache all artilces in redis
+        if articles_to_cache:
+            await cache_articles_batch(articles_to_cache)
+            print(f"Cached {len(articles_to_cache)} articles in Redis.")
+
+        return result
+
+
+async def fetch_news_by_id(article_id: str) -> Optional[Dict[str, Any]]:
```

Pylint (3.3.7)

[convention] Missing function or method docstring

(C0116)